### PR TITLE
Adding allowed http hosts flag

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

Enables limiting which HTTP Hosts are allowed in HTTP requests.

## How this works

Adds new config flag to specify the allowed hosts

## How this was tested

Added unit test

### CODE REVIEW

File: config/keys.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ const (
 	KeystoreAPIEnabledKey                              = "api-keystore-enabled"
 	MetricsAPIEnabledKey                               = "api-metrics-enabled"
 	HealthAPIEnabledKey                                = "api-health-enabled"
+	APIAllowedHostsKey                                 = "api-allowed-hosts"
```
Comment (StephenButtolph) [inline config/keys.go:147]: Should this be `"http-allowed-hosts"` and defined next to `HTTPAllowedOrigins`?

File: node/config.go [hunk 2]
```diff
@@ -54,6 +54,7 @@ type HTTPConfig struct {
 	HTTPSCert    []byte `json:"-"`
 
 	APIAllowedOrigins []string `json:"apiAllowedOrigins"`
```
Comment (joshua-kim) [inline node/config.go:56]: unrelated but I think this json tag is named weirdly because the config flag for this is actually called `http-allowed-origins`

**Review by StephenButtolph — APPROVED** [review]

File: node/config.go (hunk 2 above)
Comment (StephenButtolph) [reply node/config.go:56]: Yeah - we should add that in a separate PR

**Review by danlaine — APPROVED** [review]

File: node/config.go (hunk 2 above)
Comment (joshua-kim) [reply node/config.go:56]: PR - https://github.com/ava-labs/avalanchego/pull/1567/files

**Pull request merged**